### ISSUE TITLE
Allow skipping instruction deserialization

### ISSUE THREAD

**Issue by archseer** [project-serum/anchor#994]
I'd like to tag certain instructions as `raw` which would receive the payload as an undecoded byte slice. That way we can implement custom ser/de logic in instructions where it's too expensive to use borsh.

**Comment by armaniferrante** [comment]
You can do this with the fallback function. Example here https://github.com/project-serum/anchor/blob/master/tests/misc/programs/misc/src/lib.rs#L126-L132

**Comment by archseer** [comment]
Ah cool, I didn't know that would work 👍🏻 

Seems like the feature is undocumented, so maybe this issue can be renamed to "document fallback functions"?

**Comment by archseer** [comment]
Ah, looks like this won't handle the account context for us though. Looks like I can manually construct it via `Context::new`

**Comment by armaniferrante** [comment]
> Ah, looks like this won't handle the account context for us though. Looks like I can manually construct it via `Context::new`

Correct. It provides the raw solana entrypoint interface, which will execute if the provided instruction data doesn't contain an identifier matching any of the other instructions in the program module.

**Comment by archseer** [comment]
For my usecase, I'm trying to pass through the data as raw, but still use the Anchor context. I followed the skeleton code here: https://github.com/project-serum/anchor/blob/d1edf2653f13f908a095081ec95d4b2c85a0b2d2/lang/syn/src/codegen/program/handlers.rs#L598-L625

It's less convenient though because the generated account context isn't in the IDL so I have to manually construct the `AccountMeta` list in typescript. Could we support a signature like this?

```rust 
    pub fn default(
        ctx: Context<Default>,
        data: &[u8],
    ) -> ProgramResult {
        // ...
    }
```
This way raw instructions would still be present in the IDL and callable from typescript.

**Comment by paul-schaaf** [comment]
I like it
